templates/elixir/flox.toml

```diff
@@ -0,0 +1,45 @@
+[programs]
+elixir_1_13 = {}
+postgresql_14 = {}
+nodejs = {}
+
+[hooks]
+
+postShellHook = '''
+echo "$PWD"
+export LANG=C.UTF-8
+# keep your shell history in iex
+export ERL_AFLAGS="-kernel shell_history enabled"
+
+
+export PGDATA=$PWD/postgres_data
+export PGHOST=$PWD/postgres 
+export LOG_PATH=$PWD/postgres/LOG
+export PGDATABASE=postgres
+export DATABASE_URL="postgresql:///postgres?host=$PGHOST"
+if [ ! -d $PWD/postgres ]; then
+mkdir -p $PWD/postgres
+fi
+if [ ! -d $PGDATA ]; then
+echo 'Initializing postgresql database...'
+initdb $PGDATA --auth=trust >/dev/null
+fi
```

Comment:
This block doesn't immediately feel to me like it should belong in a postShellHook. Maybe just because I instinctively react negatively to things that are stateful. But I do wonder if we need a cleaner way to integrate with state in dev envs. Because obviously it's desirable. But not sure if there's a clean way to interact with it right now 🤔

Reply:
@mkenigs good points.

these kinds of things are typically done in shell hooks, in shell.nix files or devShells in my experience

There is an example at 17.3.7.2.2. Elixir - Phoenix project  from https://nixos.org/manual/nixpkgs/stable/#how-to-develop for the section on BEAM packaging that does use a `shellHook` to do very similar things. It is feasible to do the work outside of the shell hook. And it could be this is better at most as an example/demonstration rather than mechanism where we deterministically try to manage this for people.